src/java.base/share/classes/java/util/Objects.java

```diff
@@ -235,7 +235,7 @@ public static String toString(@GuardSatisfied @Nullable Object o, String nullDef
      * @throws NullPointerException if {@code obj} is {@code null}
      */
     @CFComment({"lock: TODO: treat like other nullness assertion methods in the Checker Framework."})
-    public static <T extends @NonNull Object> T requireNonNull(T obj) {
+    public static <T> @NonNull T requireNonNull(@NonNull T obj) {
```

Comment:
The previous annotations were correct, but resulted in `type.argument.type.incompatible` errors when null is passed.  These annotations produce `argument.type.incompatible` errors.  This makes it clear that the argument is the wrong type.